src/darwin/Framework/CHIP/MTRDeviceControllerFactory.mm

```diff
@@ -1246,6 +1260,42 @@ - (PersistentStorageDelegate *)storageDelegate
     return &_groupDataProvider;
 }
 
+#pragma mark - Controller suspension support
+
+- (BOOL)_allControllersSuspended
+{
+    auto * controllers = [self getRunningControllers];
+    for (MTRDeviceController * controller in controllers) {
+        if (!controller.suspended) {
+            return NO;
+        }
+    }
+    return YES;
+}
+
+- (void)controller:(MTRDeviceController *)controller
```

Comment:
Seems like this was added in a different PR, but this reads like the delegate is being queried whether a particular controller is suspended. Something like `controller:didChangeSuspensionState:` would be more obvious (or even two separate `controllerDidSuspend:` and `controllerDidResume:` methods).

Reply:
Understood.  I'll see about changing the names in a followup; there's coupling here with some other code, unfortunately.

Reply:
https://github.com/project-chip/connectedhomeip/pull/35577